doc/man7/EVP_KDF-ARGON2.pod

```diff
@@ -0,0 +1,169 @@
+=pod
+
+=head1 NAME
+
+EVP_KDF-ARGON2 - The Argon2 EVP KDF implementation
+
+=head1 DESCRIPTION
+
+Support for computing the B<argon2> password-based KDF through the B<EVP_KDF>
+API.
+
+The EVP_KDF-ARGON2 algorithm implements the Argon2 password-based key
+derivation function, as described in IETF RFC 9106.  It is memory-hard in
+the sense that it deliberately requires a significant amount of RAM for efficient
+computation. The intention of this is to render brute forcing of passwords on
+systems that lack large amounts of main memory (such as GPUs or ASICs)
+computationally infeasible.
+
+=head2 Supported parameters
+
+The supported parameters are:
+
+=over 4
+
+=item "pass" (B<OSSL_KDF_PARAM_PASSWORD>) <octet string>
+
+=item "salt" (B<OSSL_KDF_PARAM_SALT>) <octet string>
+
+=item "secret" (B<OSSL_KDF_PARAM_SECRET>) <octet string>
+
+Secret is an optional secret data used along the password.
+
+=item "iter" (B<OSSL_KDF_PARAM_ITER>) <unsigned integer>
+
+These parameters work as described in L<EVP_KDF(3)/PARAMETERS>.
+
+=item "threads" (B<OSSL_KDF_PARAM_THREADS>) <unsigned integer>
+
+The number of threads. This can only be used with built-in thread support.
+Threading must be explicitly enabled. See EXAMPLES section for more information.
+
+=item "size" (B<OSSL_KDF_PARAM_SIZE>) <unsigned integer>
+
+Output size.
+
+=item "ad" (B<OSSL_KDF_PARAM_ARGON2_AD>) <octet string>
+
+Optional associated data, may be used to "tag" a group of keys, or tie them
+to a particular public key, without having to modify salt.
+
+=item "lanes" (B<OSSL_KDF_PARAM_ARGON2_LANES>) <unsigned integer>
+
+Number of lanes.
```

Comment:
I'd like to see some description of relation between the `lanes` and `threads` parameter. I.e., that for maximum efficiency of the computation lanes == threads. But threads should not exceed the number of CPU cores available, otherwise it does not bring any speed up. (If I understand the Argon2 design right.)

Reply:
Yes, good